DSharpPlus.SlashCommands/SlashCommandsExtension.cs

```diff
@@ -159,6 +196,34 @@ internal Task UpdateApplicationCommandsAsync()
             return Task.CompletedTask;
         }
 
+        public void RegisterConverter<T>(ISlashArgumentConverter<T> converter, ApplicationCommandOptionType optionType) => this.RegisterConverter(typeof(T), converter, optionType);
+
+        public void RegisterConverter(Type entityType, ISlashArgumentConverter converter, ApplicationCommandOptionType optionType)
+        {
+            if (entityType is null)
+                throw new ArgumentNullException(nameof(entityType));
+            else if (converter is null)
+                throw new ArgumentNullException(nameof(converter));
+            else if (!entityType.IsEquivalentTo(converter.GetType().GenericTypeArguments[0]))
+                throw new ArgumentException($"The entity type ({entityType.FullName}) must be the same as the converter type ({converter.GetType().GenericTypeArguments[0].FullName}).", nameof(entityType));
+            this._validOptionTypes.Add(entityType, optionType);
+            this._converters.Add(entityType, converter);
+        }
+
+        public void UnregisterConverter<T>() => this.UnregisterConverter(typeof(T));
```

Comment:
this might not be something we want to support - it allows some very annoying and potentially hard to debug ways to break code. what specifically is the use case for this?

Reply:
I did this quite literally just because we can. If nobody can think of a good use case, I'll remove it